Parsing error: Cannot use keyword 'await' outside an async function

**What version of this package are you using?**

14.3.4

**What operating system, Node.js, and npm version?**

Node: 14.13.0
npm: 6.14.8
OS: macOS Catalina 10.15.5

**What happened?**

Given code:

```js
import { fileURLToPath } from 'url'
import fs from 'fs'
const file = await fs.promise.readFile(fileURLToPath(import.meta.url))
console.log(file.toString())
```

Output is:

```
standard: Use JavaScript Standard Style (https://standardjs.com)
  example.mjs:3:14: Parsing error: Cannot use keyword 'await' outside an async function
```


**What did you expect to happen?**

`Top-level await` is a stage 3 proposal (see https://tc39.es/proposal-top-level-await/) but is enabled by default in from Node 14.8.0 (see https://github.com/nodejs/node/blob/master/doc/changelogs/CHANGELOG_V14.md#2020-08-11-version-1480-current-codebytere). 

This is going to mean that linting with apps written for Node 14+ will be problematic, as top-level await is bound to be used. 

Since Node 14 is becoming Active LTS in a few weeks I think TLA should pass linting

**Are you willing to submit a pull request to fix this bug?**

yes

@voxpelli Ok, here goes my 2 cents.
Using only ESLint, if you don't set the `parserOptions.ecmaVersion` to `"latest"` it also triggers the error

```js
// .eslintrc.js

module.exports = {
  env: {
    es2021: true,
  },
  parserOptions: {
    ecmaVersion: 'latest', // <--- This line!!!
    sourceType: 'module'
  }
}
```

```js
// main.js

await new Promise(resolve => resolve())
```

With `parserOptions.ecmaVersion: "latest"`
```sh
$ eslint .\main.js
$
```

Without
```sh
$ eslint .\main.js

.\main.js
  1:1  error  Parsing error: Cannot use keyword 'await' outside an async function

✖ 1 problem (1 error, 0 warnings)

$
```

--- a/package.json
+++ b/package.json
@@ -17,7 +17,7 @@
   "dependencies": {
     "eslint": "^8.8.0",
     "eslint-config-standard": "17.0.0-1",
-    "eslint-config-standard-jsx": "11.0.0-0",
+    "eslint-config-standard-jsx": "11.0.0-1",
     "eslint-plugin-import": "^2.25.4",
     "eslint-plugin-n": "^14.0.0",
     "eslint-plugin-promise": "^6.0.0",

--- a/test/validate-config.js
+++ b/test/validate-config.js
@@ -1,0 +1,16 @@
+import test from 'tape'
+import standard from '../index.js'
+
+test('load config in eslint to validate all rule syntax is correct', async function (t) {
+  t.plan(1)
+  const code = 'const foo = 1\nconst bar = function () {}\nbar(foo)\n'
+  const [result] = await standard.lintText(code)
+  t.equal(result.errorCount, 0)
+})
+
+test('ensure we allow top level await', async function (t) {
+  t.plan(1)
+  const code = 'const foo = await 1\nconst bar = function () {}\nawait bar(foo)\n'
+  const [result] = await standard.lintText(code)
+  t.equal(result.errorCount, 0)
+})

